server/src/test/java/org/elasticsearch/ingest/PipelineProcessorTests.java

```diff
@@ -130,4 +126,76 @@ public void testAllowsRepeatedPipelineInvocations() throws Exception {
         outerProc.execute(testIngestDocument);
         outerProc.execute(testIngestDocument);
     }
+
+    public void testPipelineProcessorWithPipelineChain() throws Exception {
+        String pipeline1Id = "pipeline1";
+        String pipeline2Id = "pipeline2";
+        String pipeline3Id = "pipeline3";
+        IngestService ingestService = mock(IngestService.class);
+        PipelineProcessor.Factory factory = new PipelineProcessor.Factory(ingestService);
+
+        Map<String, Object> pipeline1ProcessorConfig = new HashMap<>();
+        pipeline1ProcessorConfig.put("pipeline", pipeline2Id);
+        PipelineProcessor pipeline1Processor = factory.create(Collections.emptyMap(), null, pipeline1ProcessorConfig);
+
+        Map<String, Object> pipeline2ProcessorConfig = new HashMap<>();
+        pipeline2ProcessorConfig.put("pipeline", pipeline3Id);
+        PipelineProcessor pipeline2Processor = factory.create(Collections.emptyMap(), null, pipeline2ProcessorConfig);
+
+        Pipeline pipeline1 = new Pipeline(
+            pipeline1Id, null, null, new CompoundProcessor(pipeline1Processor)
+        );
+
+        String key1 = randomAlphaOfLength(10);
+        Pipeline pipeline2 = new Pipeline(
+            pipeline2Id, null, null, new CompoundProcessor(true,
+            Arrays.asList(
+                new TestProcessor(ingestDocument -> {
+                    ingestDocument.setFieldValue(key1, randomInt());
+                    try {
+                        Thread.sleep(100); //force the stat time to be non-zero
```

Comment:
Shouldn't it technically suffice to sleep for `2` here (or even `1`, not sure about the potential rounding/accuracy issues with that and `nanotime()`, but `2` should be fine for getting a visible change in the metric?)

Reply:
yeah, 100 is bit overkill. I will drop this down.